kernel/mmanutils.c

```diff
@@ -1148,6 +1148,15 @@ ssize_t myst_mman_pids_test(const void* addr, size_t length, pid_t pid)
     return _handle_mman_pids_op(MMAN_PIDS_OP_TEST, addr, length, pid);
 }
 
+bool myst_is_within_process(const void* addr, pid_t pid)
+{
+    uint64_t addr_page = myst_round_down_to_page_size((uint64_t)addr);
```

Comment:
How would `myst_is_within_process` handle shared memory?

Reply:
the updated PR is now simply checking if the page is mapped or not instead of the ownership.